go/libraries/doltcore/sqle/tables.go

```diff
@@ -3162,99 +3150,69 @@ func (t *AlterableDoltTable) CreateCheck(ctx *sql.Context, check *sql.CheckDefin
 		return err
 	}
 
-	table, err := t.DoltTable.DoltTable(ctx)
-	if err != nil {
-		return err
-	}
-
-	newTable, err := table.UpdateSchema(ctx, sch)
-	if err != nil {
-		return err
-	}
-
-	newRoot, err := root.PutTable(ctx, t.TableName(), newTable)
-	if err != nil {
-		return err
-	}
-
-	err = t.setRoot(ctx, newRoot)
-	if err != nil {
-		return err
-	}
-
-	return t.updateFromRoot(ctx, newRoot)
+	return t.updateSchema(ctx, root, sch)
 }
 
 func (t *AlterableDoltTable) DropCheck(ctx *sql.Context, chName string) error {
-	if err := dsess.CheckAccessForDb(ctx, t.db, branch_control.Permissions_Write); err != nil {
-		return err
-	}
-	root, err := t.getRoot(ctx)
+	root, sch, err := t.getWritableSchema(ctx)
 	if err != nil {
 		return err
 	}
 
-	updatedTable, _, err := root.GetTable(ctx, t.TableName())
+	err = sch.Checks().DropCheck(chName)
 	if err != nil {
 		return err
 	}
 
-	sch, err := updatedTable.GetSchema(ctx)
-	if err != nil {
-		return err
-	}
+	return t.updateSchema(ctx, root, sch)
+}
 
-	err = sch.Checks().DropCheck(chName)
-	if err != nil {
-		return err
-	}
+func (t *AlterableDoltTable) ModifyStoredCollation(ctx *sql.Context, collation sql.CollationID) error {
+	return fmt.Errorf("converting the collations of columns is not yet supported")
+}
 
-	table, err := t.DoltTable.DoltTable(ctx)
+func (t *AlterableDoltTable) ModifyDefaultCollation(ctx *sql.Context, collation sql.CollationID) error {
+	root, sch, err := t.getWritableSchema(ctx)
 	if err != nil {
 		return err
 	}
 
-	newTable, err := table.UpdateSchema(ctx, sch)
-	if err != nil {
-		return err
-	}
+	sch.SetCollation(schema.Collation(collation))
 
-	newRoot, err := root.PutTable(ctx, t.TableName(), newTable)
-	if err != nil {
-		return err
-	}
+	return t.updateSchema(ctx, root, sch)
+}
 
-	err = t.setRoot(ctx, newRoot)
+func (t *AlterableDoltTable) ModifyComment(ctx *sql.Context, comment string) error {
+	root, sch, err := t.getWritableSchema(ctx)
 	if err != nil {
 		return err
 	}
 
-	return t.updateFromRoot(ctx, newRoot)
-}
+	sch.SetComment(comment)
 
-func (t *AlterableDoltTable) ModifyStoredCollation(ctx *sql.Context, collation sql.CollationID) error {
-	return fmt.Errorf("converting the collations of columns is not yet supported")
+	return t.updateSchema(ctx, root, sch)
 }
 
-func (t *AlterableDoltTable) ModifyDefaultCollation(ctx *sql.Context, collation sql.CollationID) error {
+func (t *AlterableDoltTable) getWritableSchema(ctx *sql.Context) (doltdb.RootValue, schema.Schema, error) {
```

Comment:
Would be nice to give this method a godoc string. It's mostly intuitive from the name, but would be nice to call out things like that this method will check for write access permissions, since that's not obvious from the name.